### PR TITLE
Expand example documentation

### PR DESCRIPTION
**Pull request by timClicks**
## Motivation and Context

The current README for the example services is fairly spartan.

## Description

I've expanded the README, specifically:

- Clean up some markup
- Document pre-reqs.
- Document all make targets.
- Avoid the $BINARY because it looks like a bash variable.
- Expand intro

## Testing

n/a

## Checklist

n/a

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,49 +1,89 @@
 # Smithy Rust Server SDK examples
 
-This folder contains an example services showcasing the service framework capabilities and to run benchmarks.
+This folder contains an example services showcasing Smithy Rust Server SDK,
```
Comment (crisidev) [inline examples/README.md:3]:
```suggestion
This folder contains some example services showcasing Smithy Rust Server SDK, 
```

File: examples/README.md [hunk 2]
```diff
@@ -1,49 +1,89 @@
...
 
-Since this example requires both the server and client SDK to be code-generated
+[middleware]: https://awslabs.github.io/smithy-rs/design/server/middleware.html
+[extractors]: https://awslabs.github.io/smithy-rs/design/server/from_parts.html
+
+
+## Pre-requisites
+
+You will need install Java 11 (or higher) and [Gradle](https://gradle.org/) 7 
+(or higher) to run the smithy-rs code generator. You will need a working
```
Comment (crisidev) [inline examples/README.md:24]: Gradle is provisioned automatically at the right version when you run `gradlew` the first time.

Comment (rcoh) [reply examples/README.md:24]: Also, the compatibility range is actually limited—I'd probably say "Java 11" and be done—it works with a few other versions but not all versions

File: examples/README.md [hunk 3]
```diff
@@ -1,49 +1,89 @@
...
-Since this example requires both the server and client SDK to be code-generated
+[middleware]: https://awslabs.github.io/smithy-rs/design/server/middleware.html
+[extractors]: https://awslabs.github.io/smithy-rs/design/server/from_parts.html
+
+
+## Pre-requisites
+
+You will need install Java 11 (or higher) and [Gradle](https://gradle.org/) 7 
+(or higher) to run the smithy-rs code generator. You will need a working 
+installation of Rust, including `cargo`, to compile the generated code.
```
Comment (timClicks) [inline examples/README.md:25]:
```suggestion
You will need install Java 11 to run the smithy-rs code generator and 
installation of Rust, including `cargo`, to compile the generated code.
```

File: examples/README.md (hunk 2 above)
Comment (timClicks) [reply examples/README.md]: Thanks for taking a look, both. Fixed here https://github.com/awslabs/smithy-rs/pull/2570/commits/6f43a853537d2ac3d2998303f84677a2c42a5d77

**Review by hlbarber — APPROVED** [review]
Thank you kindly

**Review by crisidev — APPROVED** [review]

**Pull request merged**